### DISCUSSION TITLE
Endgame Integration

### DISCUSSION THREAD

**Discussion by welch27330** [Security-Onion-Solutions/securityonion#7234] [General]
I followed the docs and I still am not able to see alerts or data from Endgame. I verified that the data is being sent and received via tcpdump but no luck. Any assistance will be greatly appreciated.

**Comment by dougburks** [comment]
How is your Security Onion installation configured (eval, standalone, distributed)?

**Comment by welch27330** [comment]
It is a distributed deployment.

Jonathan
________________________________

**Comment by dougburks** [comment]
Which node are you sending Endgame logs to?

**Comment by welch27330** [comment]
I am sending them to my Master node.

Jonathan
________________________________

**Comment by dougburks** [comment]
Is Endgame correctly trusting the Security Onion certificate?

Does Endgame show any errors or provide any additional clues?

**Reply by welch27330** [reply]
The certificate is trusted and it shows a good connection. I can see the data coming across the interface on my master with tcpdump from the endgame server.

**Comment by dougburks** [comment]
Have you checked `/opt/so/log` for additional clues?

**Reply by welch27330** [reply]
Checked the logs, nothing relevant or I'm not looking in the right place

**Comment by dougburks** [comment]
From https://docs.securityonion.net/en/2.3/endgame.html#configuration, did you use the `During Security Onion Setup` option or the `Post-Installation Setup` option? If post-installation, did you replace `$smpip` with the actual IP address? Have you verified that the firewall rules are correct?

**Reply by welch27330** [reply]
I used the Post example and ensured that the FW rules are correct as well I verified the IP address.

**Reply by welch27330** [reply]
the traffic is for sure hitting the interface on the master

**Comment by weslambert** [comment]
Try checking the pipeline on the manger to ensure events are hitting the Logstash HTTP input for Endgame events (check the `out` value):

`sudo so-logstash-pipeline-stats manager | grep -A5 -B5 endgame`

If events are hitting the external interface, but aren’t hitting the input, it could be an issue with the local firewall rules with respect to the Docker interface, or particular iptables chains, etc.

Thanks,
Wes

**Reply by welch27330** [reply]
Wes, you called it, now what do I do? It's not hitting the pipeline

**Reply by weslambert** [reply]
What is the output of the following from the manger, redacting IPs as necessary?

`sudo iptables -nvL | grep 3765`

**Reply by welch27330** [reply]
0 0 ACCEPT tcp -- !docker0 docker0 0.0.0.0/0 172.17.0.19 tcp dpt:3765
237k 2500M ACCPEPT tcp * * </IP ADDRESS Endgame/> 0.0.0.0/0 tcp dpt:3756

**Comment by weslambert** [comment]
Did you perform pre or post-setup configuration? 

You'll want to compare the rules to something like that of Beats (5044).  

For example:

```
    0     0 ACCEPT     tcp  --  !docker0 docker0  0.0.0.0/0            172.17.0.29          tcp dpt:5044
    0     0 ACCEPT     tcp  --  *      *       192.168.6.10          0.0.0.0/0            tcp dpt:5044
    0     0 ACCEPT     tcp  --  *      *       192.168.6.100         0.0.0.0/0            tcp dpt:5044
    0     0 ACCEPT     tcp  --  *      *       127.0.0.1            0.0.0.0/0            tcp dpt:5044
```

I would double-check your SO firewall configuration with:

`sudo so-firewall includedhosts endgame`

If you don't see your SMP address, you can include it in the host group with:

`sudo so-firewall includehost endgame $smpip`

**Reply by welch27330** [reply]
I performed the post installation configuration steps as described in the guide.

**Comment by btdewey** [comment]
We just ran into this in a distributed deployment.  The index filter is `Endgame-*` which won't search any additional search nodes and the endgame data is likely on a search node.  We created a new index filter of `*:Endgame-*` and then manually modified the visualizations.

We found that data was getting into the system properly, but it was just the visualizations that were not working.  Checking indexes and pipeline data on the master node did not return good value.  Had to look at the search node before the indexes showed up.

**Reply by btdewey** [reply]
Just now realized that the index pattern on the endgame documentation page shows the correct on.  I thought we were on 2.3.110, but now I need to verify that since that index pattern should have been correct.
